### ISSUE TITLE
no-unused-variable interferes with other rules

### ISSUE THREAD

**Issue by kevinphelps** [palantir/tslint#3455]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
import { EventEmitter } from '@angular/core';
import { Observable } from 'rxjs/Observable';
import { Subject } from 'rxjs/Subject';
import 'rxjs/add/observable/of';
import 'rxjs/add/operator/do';

export const f1 = () => returnsVoid(); // error
export const obs1 = Observable.of(undefined).do(() => returnsVoid()); // error

export class Component {
  subject = new Subject<void>();
  event = new EventEmitter<void>();

  obs2 = Observable.of(undefined).do(() => this.subject.next()); // error
  obs3 = Observable.of(undefined).do(() => this.event.next()); // no error, but should be

  obs4 = Observable.of(undefined).do(this.subject.next); // error
  obs5 = Observable.of(undefined).do(this.event.next); // no error, but should be
}

function returnsVoid() { }

```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-unused-variable": [
      true,
      "check-parameters", {
        "ignore-pattern": "^_"
      }
    ],
    "no-unbound-method": true,
    "no-void-expression": true
  }
}
```

#### Actual behavior
When the `no-unused-variable` variable is enabled, the `no-void-expression` rule does not report an
error on line 15 and the `no-unbound-method` rule does not report an error on line 18. But if you
disable the `no-unused-variable` rule, the `no-void-expression` and `no-unbound-method` rules
correctly reports all of the noted errors.

#### Expected behavior
The `no-unused-variable` rule should not interfere with other rules.

**Comment by kevinphelps** [comment]
Repro: https://github.com/kevinphelps/no-void-expression-bug

```shell
git clone  https://github.com/kevinphelps/no-void-expression-bug.git
cd no-void-expression-bug
yarn
yarn run lint
```

**Comment by kevinphelps** [comment]
There actually seems to be more to this. In my larger project, `no-void-expression` misses errors when `no-unused-variable` is enabled regardless of the order. I will try to isolate that into the repro.

**Comment by kevinphelps** [comment]
`no-unused-variable` appears to also interfere is with other rules like `no-unbound-method`.

**Comment by kevinphelps** [comment]
(did not mean to close the issue)

**Comment by ajafff** [comment]
Duplicate of #2649 and #2736
Fix is available in #2763 but unlikely to be merged soon (if at all)

**Comment by kevinphelps** [comment]
Sorry about that...

**Comment by ajafff** [comment]
Don't worry about that. The other issues were hard to find.
I just renamed one of them to make it easier to find.

**Comment by kevinphelps** [comment]
I saw that. I thought I was an idiot until I saw the rename.

**Comment by kevinphelps** [comment]
I would love to just enable the proper TypeScript settings and just remove `no-unused-any`, but I can't do that until I upgrade my project to Angular 5 to ngc issues. So my temporary workaround is to run the `no-unused-any` rule by itself and then run the rest in a separate process.

**Comment by ajafff** [comment]
You mean the `Parameter '$event' is declared but never used.` error when compiling AOT?
You just need to disable `--noUnusedParameters` while still using `--noUnusedLocals`.
To still find unused parameters you can use the `no-unused` rule from my package `tslint-consistent-codestyle`: https://github.com/ajafff/tslint-consistent-codestyle/blob/master/docs/no-unused.md

**Comment by kevinphelps** [comment]
I'll give that a shot! Thanks!
